### PR TITLE
[Synapse Data Plane]Add Synapse role and role assignment swagger

### PR DESCRIPTION
**Pull request by aim-for-better**
Swagger Spec for Role Assignment and Role APIs
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+        "operationId": "RoleAssignments_CreateRoleAssignment",
+        "description": "Create role assignment.",
+        "x-ms-examples": {
+          "Create role assignment": {
+            "$ref": "./examples/RoleAssignments_CreateRoleAssignment.json"
+          }
+        },
+        "consumes": [
+          "application/json",
+          "text/json"
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:41]: I would remove "text/json", that is not a recommended mimetype and was in old Swagger for compat reason, I can't suggest it on new Swagger

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        ],
+        "produces": [
+          "application/json",
+          "text/json"
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:45]: same

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/RoleAssignmentDetails"
+      }
+    },
+    "RoleAssignmentRequest": {
+      "description": "Role Assignment request details",
+      "type": "object",
+      "properties": {
+        "roleId": {
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:269]: are any of those parameters required?

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success response.",
+            "schema": {
+              "$ref": "#/definitions/RolesListResponse"
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json:53]: This operation fits the condition of the x-ms-pageable extension, please use it:
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json (hunk 2 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:45]: > same

Removed

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json (hunk 1 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:41]: > I would remove "text/json", that is not a recommended mimetype and was in old Swagger for compat reason, I can't suggest it on new Swagger

Removed

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json (hunk 3 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:269]: > are any of those parameters required? right now they are both optional. The operation parameter requires the class, but the class has two optional parameters. You migth want to add
> 
> ```
> required: [
>   roleId
>   principalId
> ]
> ```
> 
> If they are both required.

Fixed. Add required

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json (hunk 4 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json:53]: > This operation fits the condition of the x-ms-pageable extension, please use it:
> https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable

Added x-ms-pageable

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+        "description": "List role assignments.",
+        "x-ms-examples": {
+          "Get role assignment information": {
+            "$ref": "./examples/RoleAssignments_ListRoleAssignments.json"
+          }
+        },
+        "consumes": [],
+        "produces": [
+          "application/json",
+          "text/json"
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json:88]: Sorry I meant to remove them all, I thought I was clear sorry :(

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        "description": "List roles.",
+        "x-ms-examples": {
+          "Get access control information": {
+            "$ref": "./examples/Roles_ListRoles.json"
+          }
+        },
+        "consumes": [],
+        "produces": [
+          "application/json",
+          "text/json"
```
Comment (lmazuel) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json:42]: Same remove them all

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json (hunk 5 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roleAssignments.json]: > Sorry I meant to remove them all, I thought I was clear sorry :(

Sorry, I have removed all text/json

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json (hunk 6 above)
Comment (aim-for-better) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/roles.json]: > Same remove them all

removed

**Review by lmazuel — APPROVED** [review]

**Pull request merged**